internal/surfer/gcloud/builder.go

```diff
@@ -0,0 +1,403 @@
+	hostParts := strings.Split(service.DefaultHost, ".")
+	shortServiceName := hostParts[0]
+
+	// We determine the singular name of the resource.
+	// For `Create` methods, this comes from the `_id` field. For others, it's the `name` field.
+	resourceName := strcase.ToSnake(strings.TrimSuffix(field.Name, "_id"))
+	if field.Name == "name" {
+		resourceName = utils.GetSingularFromSegments(segments)
+	}
+
+	// We generate a helpful help text based on whether the command is a `Create` command or not.
+	helpText := fmt.Sprintf("The %s to create.", resourceName)
+	if !strings.HasPrefix(method.Name, "Create") {
+		helpText = fmt.Sprintf("The %s to operate on.", resourceName)
+	}
+
+	// We assemble the final `Param` struct with all the necessary information for a primary resource.
+	return Param{
+		HelpText:          helpText,
+		IsPositional:      true,
+		IsPrimaryResource: true,
+		Required:          true,
+		RequestIDField:    strcase.ToLowerCamel(field.Name),
+		ResourceSpec: &ResourceSpec{
+			Name:                  resourceName,
+			PluralName:            utils.GetPluralResourceNameForMethod(method, model),
+			Collection:            fmt.Sprintf("%s.%s", shortServiceName, collectionPath),
+			DisableAutoCompleters: false,
+			Attributes:            newAttributesFromSegments(segments),
+		},
+	}
+}
+
+// newResourceReferenceSpec creates a ResourceSpec for a field that references
+// another resource type (e.g., a `--network` flag).
+func newResourceReferenceSpec(field *api.Field, model *api.API, _ *Config, service *api.Service) (*ResourceSpec, error) {
+	// We iterate through all the resource definitions in the API model to find the
+	// one that matches the type of our resource reference.
+	for _, def := range model.ResourceDefinitions {
+		if def.Type == field.ResourceReference.Type {
+			if len(def.Patterns) == 0 {
+				return nil, fmt.Errorf("resource definition for %q has no patterns", def.Type)
+			}
+			segments := def.Patterns[0]
+
+			// We determine the plural name, using the explicit `plural` field if available,
+			// and falling back to parsing the pattern otherwise.
+			pluralName := def.Plural
+			if pluralName == "" {
+				pluralName = utils.GetPluralFromSegments(segments)
+			}
+
+			// We determine the singular name from the pattern.
+			name := utils.GetSingularFromSegments(segments)
+
+			// We construct the full gcloud collection path for the referenced resource
+			//assuming the current service is the current command.
+			hostParts := strings.Split(service.DefaultHost, ".")
+			shortServiceName := hostParts[0]
+			baseCollectionPath := utils.GetCollectionPathFromSegments(segments)
+			fullCollectionPath := fmt.Sprintf("%s.%s", shortServiceName, baseCollectionPath)
+
+			// We assemble and return the `ResourceSpec`.
+			return &ResourceSpec{
+				Name:                  name,
+				PluralName:            pluralName,
+				Collection:            fullCollectionPath,
+				DisableAutoCompleters: true,
```

Comment:
The hard coded `true` I don't think is quite right

Reply:
https://github.com/googleapis/librarian/issues/3416